src/Compiler/Checking/MethodOverrides.fs

```diff
@@ -379,8 +379,9 @@ module DispatchSlotChecking =
                     // error reporting path
                     let compiledSig = CompiledSigOfMeth g amap m dispatchSlot
                     
-                    let noimpl() = 
-                        missingOverloadImplementation.Add((isReqdTyInterface, lazy NicePrint.stringOfMethInfo infoReader m denv dispatchSlot))
+                    let noimpl() =
+                        if (isObjExpr && dispatchSlot.IsInstance) || dispatchSlot.IsInstance then
```

Comment:
Isn't this just
```suggestion
                        if dispatchSlot.IsInstance then
```
?

Reply:
Yep. I think that work too :)

Reply:
But to be completely honest I am not sure what it was supposed to be 🙃 Will it now check or skip if interface is implemented this way https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/interfaces#implementing-interfaces-by-using-class-types (and is it supposed to check or not? I think it is supposed to still check)?

I was merely pointing out that `(A && B) || B` is equivalent to `B`, but maybe it was supposed to be `A && B`?

Reply:
Sorry, was a bit too laconic with my first comment

Reply:
> But to be completely honest I am not sure what it was supposed to be 🙃 Will it now check or skip if interface is implemented this way https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/interfaces#implementing-interfaces-by-using-class-types (and is it supposed to check or not? I think it is supposed to still check)?
> 
> I was merely pointing out that `(A && B) || B` is equivalent to `B`, but maybe it was supposed to be `A && B`?

The fact that this code is called from different places,  I added the long check to prevent from breaking cases that were not tested. Lets see what the CI says. I will add more tests nevertheless

Reply:
I think we should skip the check for static overrides when Implementing interfaces by using class types and raise a warning/error when using static members in the implementation. Unless they are part of IWSAM. cc @vzarytovskii